app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/CreateAccountActivity.java

```diff
@@ -0,0 +1,180 @@
+    private TextInputLayout phoneLayout;
+    private TextInputEditText phoneField;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create);
+        this.getSupportActionBar().hide();
+        // hides the status bar, deprecated in API 30
+        View decorView = this.getWindow().getDecorView();
+        int uiOptions = View.SYSTEM_UI_FLAG_FULLSCREEN;
+        decorView.setSystemUiVisibility(uiOptions);
+
+        usernameLayout = findViewById(R.id.login_username_layout);
+        usernameField = (TextInputEditText) usernameLayout.getEditText();
+        usernameField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable usernameString) {
+                // remove the error if any exist
+                usernameLayout.setError(null);
+            }
+        });
+
+        passwordLayout = findViewById(R.id.login_password_layout);
+        passwordField = (TextInputEditText) passwordLayout.getEditText();
+        passwordField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable passwordString) {
+                // remove the error if any exist
+                passwordLayout.setError(null);
+            }
+        });
+
+        confirmLayout = findViewById(R.id.confirm_password_layout);
+        confirmField = (TextInputEditText)confirmLayout.getEditText();
+        confirmField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable comfirmString) {
+                // remove the error if any exist
+                confirmLayout.setError(null);
+            }
+        });
+
+        emailLayout = findViewById(R.id.email_layout);
+        emailField = (TextInputEditText)emailLayout.getEditText();
+        emailField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable emailString) {
+                // remove the error if any exist
+                emailLayout.setError(null);
+            }
+        });
+
+        phoneLayout = findViewById(R.id.phone_layout);
+        phoneField = (TextInputEditText) phoneLayout.getEditText();
+        phoneField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable phoneNumber) {
+                // remove the error if any exist
+                phoneLayout.setError(null);
+            }
+        });
+
+    }
+    /**
+     * check the username field for emptiness, adding error message if field is empty
+     * @param username the username that the user entered
+     * @return true if the username is not empty, false if username is empty
+     */
+    private boolean checkUsername(String username) {
```

Comment:
I think all these checks can be merge into one called checkEmpty.
We can take in a TextInputLayout parameter and an error message and extract text here to check.
`getString(R.string.string_username) + getString(R.string.empty_error)` this won't work in this approach, we might want to have a different error message for each inputLayout.